controllers/util/common/common.go

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2022 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/IBM/ibm-block-csi-operator/controllers/internal/controller_instance"
+	"github.com/IBM/ibm-block-csi-operator/controllers/util"
+	oconfig "github.com/IBM/ibm-block-csi-operator/pkg/config"
+	"github.com/go-logr/logr"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ControllerHelper struct {
+	client.Client
+	Logger logr.Logger
```

Comment:
we need to pick, [Log](https://github.com/IBM/ibm-block-csi-operator/blob/54972b354d510ef07ea46b6d3aecaef71cbf6af6/controllers/ibmblockcsi_controller.go#L71) or Logger but not both.

Reply:
and why [this](https://github.com/IBM/ibm-block-csi-operator/blob/54972b354d510ef07ea46b6d3aecaef71cbf6af6/controllers/ibmblockcsi_controller.go#L62) way is not enough?

Reply:
done, changed it to Log

Reply:
what about the global?

Reply:
I removed [this log](https://github.com/IBM/ibm-block-csi-operator/blob/54972b354d510ef07ea46b6d3aecaef71cbf6af6/controllers/ibmblockcsi_controller.go#L71), and I think passing the [log](https://github.com/IBM/ibm-block-csi-operator/blob/54972b354d510ef07ea46b6d3aecaef71cbf6af6/controllers/ibmblockcsi_controller.go#L62) is good because it doesn't change the logs structure that was before the change